src/injected/web/gm-api.js

```diff
@@ -0,0 +1,220 @@
+import { cache2blobUrl, getUniqId } from '#/common';
+import bridge from './bridge';
+import store from './store';
+import { onTabCreate } from './tabs';
+import { onRequestCreate } from './requests';
+import { onDownload } from './download';
+import { onNotificationCreate } from './notifications';
+import {
+  decodeValue, dumpValue, loadValues, changeHooks,
+} from './gm-values';
+import {
+  findIndex, indexOf, slice, objectKeys, objectValues, objectEntries,
+  atob, Error, jsonDump, log, utf8decode,
+} from '../utils/helpers';
+
+const { getElementById } = Document.prototype;
+const { lastIndexOf } = String.prototype;
+const { hasOwnProperty } = Object.prototype;
+
+export function createGmApiProps() {
+  // these are bound to script data that we pass via |this|
+  const boundProps = {
+    GM_deleteValue(key) {
+      const { id } = this;
```

Comment:
I prefer passing an context object explicitly, like what *koa v2* did, so that we don't have to concern about function bindings any more.
Is there any advantage to use `this` rather than pass an object explicitly?

Reply:
I made an attempt to implement it explicitly as I knew you wouldn't like `this`, however it looked clumsy, un-JavaScript-like, actually Python-like.

Reply:
It's not like I have a strong opinion on this, so do you insist? :-)

Reply:
Alright, it does not matter since all the functions are called with `.apply(selfData, args)`.

Reply:
FWIW we can call all of them as gmFunction(context, ...args).
Like I said I'm not exactly *against* the more verbose approach.